packages/nc-gui-v2/pages/index/index/create-external.vue

```diff
@@ -77,51 +93,91 @@ const validators = computed(() => {
 
 const { validate, validateInfos } = useForm(formState, validators)
 
+const populateName = (v: string) => {
+  formState.dataSource.connection.database = `${v.trim()}_noco`
+}
+
 const onClientChange = () => {
   formState.dataSource = { ...getDefaultConnectionConfig(formState.dataSource.client) }
+  populateName(formState.title)
 }
 
-const inflectionTypes = ['camelize', 'none']
-const configEditDlg = ref(false)
+const onSSLModeChange = ((mode: 'No' | 'Allow' | string) => {
+  switch (mode) {
+    case 'No':
+      delete formState.dataSource.connection.ssl
+      break
+    case 'Allowed':
+      formState.dataSource.connection.ssl = 'true'
+      break
+    default:
+      formState.dataSource.connection.ssl = {
+        ca: '',
+        cert: '',
+        key: '',
+      }
+      break
+  }
+}) as unknown as SelectHandler
 
-// populate database name based on title
-watch(
-  () => formState.title,
-  (v) => (formState.dataSource.connection.database = `${v?.trim()}_noco`),
-)
+const updateSSLUse = () => {
+  if (formState?.dataSource?.connection?.ssl) {
```

Comment:
Unnecessary `?`
`formState` should not be undefined
`dataSource` can be undefined? Should probably not be undefined either so the check might be unnecessary as well.
`connection` should probably be defined as well but that one I didn't check.